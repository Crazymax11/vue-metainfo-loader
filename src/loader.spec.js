/* eslint-env jest */
const loader = require('./loader.js');

describe('loader', () => {
  it('should be a function', () => {
    expect(typeof loader).toEqual('function');
  });

  it('should extract component name from filename, when resourcePath is filename', () => {
    expect(
      loader.call(
        {
          resourcePath: 'componentName.vue',
        },
        `<script>export default {}</script>`,
      ),
    ).toMatchSnapshot();
  });

  it('should extract component name from filename, when resourcePath is absolute path', () => {
    expect(
      loader.call(
        {
          resourcePath: '/data/path/componentName.vue',
        },
        `<script>export default {}</script>`,
      ),
    ).toMatchSnapshot();
  });

  it('should console error and return code as is if error occurs', () => {
    const oldError = console.error;
    const oldWarn = console.warn;

    const errorSpy = jest.fn();
    const warnSpy = jest.fn();
    console.error = errorSpy;
    console.warn = warnSpy;

    const result = loader.call(
      {
        resourcePath: 'componentName',
      },
      'export default {}',
    );
    console.error = oldError;
    console.warn = oldWarn;

    expect(result).toEqual('export default {}');
    expect(warnSpy).toHaveBeenCalledWith(
      'vue-metainfo-loader: Error has occured while processing componentName: Unexpected token (1:0)',
    );
    expect(errorSpy.mock.calls[0][0].message).toEqual('Unexpected token (1:0)');
  });

  it('should not crash component if it already has name or meta', () => {
    expect(
      loader.call(
        {
          resourcePath: 'componentName.vue',
        },
        `<script>export default {name: 'test', meta: 'test2'}</script>`,
      ),
    ).toMatchSnapshot();
  });
});
