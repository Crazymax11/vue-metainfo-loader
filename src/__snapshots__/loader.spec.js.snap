// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader should extract component name from filename, when resourcePath is absolute path 1`] = `
"<script>export default {
  meta: {
    \\"events\\": [],
    \\"props\\": [],
    \\"customTypes\\": [],
    \\"name\\": \\"componentName\\"
},
  }</script>"
`;

exports[`loader should extract component name from filename, when resourcePath is filename 1`] = `
"<script>export default {
  meta: {
    \\"events\\": [],
    \\"props\\": [],
    \\"customTypes\\": [],
    \\"name\\": \\"componentName\\"
},
  }</script>"
`;

exports[`loader should not crash component if it already has name or meta 1`] = `
"<script>export default {
  meta: {
    \\"events\\": [],
    \\"props\\": [],
    \\"customTypes\\": [],
    \\"name\\": \\"componentName\\"
},
  name: 'test', meta: 'test2'}</script>"
`;
